### PR TITLE
Add kyverno policy exception

### PR DESCRIPTION
**Pull request by QuentinBisson**
Closes https://github.com/giantswarm/giantswarm/issues/28414

### CODE REVIEW

**Comment by QuentinBisson** [comment]
image

**Comment by QuentinBisson** [comment]
Upgrading to 3.14.0 to ensure we are not elevating privileges https://github.com/DataDog/helm-charts/compare/datadog-3.13.0...datadog-3.14.0

**Comment by QuentinBisson** [comment]
Tested with in use configmap

File: helm/datadog/templates/kyverno-policy-exception.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
+{{- if .Values.kyvernoPolicyExceptions.enabled }}
+{{- if .Capabilities.APIVersions.Has "kyverno.io/v2alpha1/PolicyException" -}}
+apiVersion: kyverno.io/v2alpha1
+kind: PolicyException
+metadata:
+  annotations:
+    "helm.sh/hook": "pre-install,pre-upgrade"
+  labels:
+    {{ include "datadog.labels" . | nindent 4 }}
+  name: {{ template "datadog.fullname" . }}-cluster-agent
```
Comment (marieroque) [inline helm/datadog/templates/kyverno-policy-exception.yaml:10]: Could we add `-exception` suffix to the resource name ?

File: helm/datadog/templates/kyverno-policy-exception.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        - {{ template "datadog.fullname" . }}-cluster-agent*
+---
+apiVersion: kyverno.io/v2alpha1
+kind: PolicyException
+metadata:
+  annotations:
+    "helm.sh/hook": "pre-install,pre-upgrade"
+  labels:
+    {{ include "datadog.labels" . | nindent 4 }}
+  name: {{ template "datadog.fullname" . }}
```
Comment (marieroque) [inline helm/datadog/templates/kyverno-policy-exception.yaml:49]: Could we add `-exception` suffix to the resource name ?

File: helm/datadog/templates/kyverno-policy-exception.yaml (hunk 1 above)
Comment (QuentinBisson) [reply helm/datadog/templates/kyverno-policy-exception.yaml:10]: Why? the kind is self explanatiory right?

Comment (marieroque) [reply helm/datadog/templates/kyverno-policy-exception.yaml:10]: Well, I thought it was a best practice, for instance:
```
[datadog-app]$ k get policyexceptions.kyverno.io -A
NAMESPACE     NAME                                                         AGE
giantswarm    chart-operator-policy-exceptions                             7d22h
giantswarm    falco-falco-exceptions                                       8d
giantswarm    falco-falco-exporter-exceptions                              8d
giantswarm    falco-falcosidekick-exceptions                               8d
giantswarm    fluent-logshipping-app                                       26h
giantswarm    kyverno-app-chart-operator-policy-exception                  9d
giantswarm    macropower-analytics-panel-server                            26h
giantswarm    prometheus-meta-operator                                     39m
giantswarm    promxy-app                                                   26h
giantswarm    silence-operator                                             26h
giantswarm    silence-operator-sync                                        26h
giantswarm    trivy-exception                                              9d
kube-system   etcd-kubernetes-resources-count-exporter-policy-exceptions   9d
```

Comment (QuentinBisson) [reply helm/datadog/templates/kyverno-policy-exception.yaml:10]: Sure :(

**Comment by QuentinBisson** [comment]
@marieroque not sure why lint is failing. I think this is a circle ci issue TBH

**Comment by QuentinBisson** [comment]
SO we do need the Chart.lock for kube-linter to pass

**Pull request merged**